graphql_compiler/schema_generation/sqlalchemy/edge_descriptors.py

```diff
@@ -1,64 +1,120 @@
-        )
+        if isinstance(edge_descriptor, DirectEdgeDescriptor):
+            from_column = edge_descriptor.from_column
+            to_column = edge_descriptor.to_column
+            join_descriptors[edge_descriptor.from_vertex][out_edge_name] = DirectJoinDescriptor(
+                from_column, to_column
+            )
+            join_descriptors[edge_descriptor.to_vertex][in_edge_name] = DirectJoinDescriptor(
+                to_column, from_column
+            )
+        elif isinstance(edge_descriptor, CompositeEdgeDescriptor):
+            join_descriptors[edge_descriptor.from_vertex][out_edge_name] = CompositeJoinDescriptor(
+                edge_descriptor.column_pairs
+            )
+            join_descriptors[edge_descriptor.to_vertex][in_edge_name] = CompositeJoinDescriptor(
+                {
+                    (to_column, from_column)
+                    for from_column, to_column in edge_descriptor.column_pairs
+                }
+            )
+        else:
+            raise AssertionError(
+                f"Unknown edge descriptor type {edge_descriptor}: "
+                f"{type(edge_descriptor)} for edge {edge_name}."
+            )
     return join_descriptors
 
 
 def validate_edge_descriptors(
-    vertex_name_to_table: Dict[str, Table], direct_edges: Dict[str, DirectEdgeDescriptor]
+    vertex_name_to_table: Dict[str, Table], edges: Dict[str, EdgeDescriptor]
 ) -> None:
     """Validate that the edge descriptors do not reference non-existent vertices or columns."""
     # TODO(pmantica1): Validate that columns in a direct SQL edge have comparable types.
     # TODO(pmantica1): Validate that columns don't have types that probably shouldn't be used for
     #                  joins, (e.g. array types).
-    for edge_name, direct_edge_descriptor in six.iteritems(direct_edges):
-        for vertex_name, column_name in (
-            (direct_edge_descriptor.from_vertex, direct_edge_descriptor.from_column),
-            (direct_edge_descriptor.to_vertex, direct_edge_descriptor.to_column),
-        ):
+    for edge_name, edge_descriptor in six.iteritems(edges):
+        if isinstance(edge_descriptor, DirectEdgeDescriptor):
+            vertex_column_pairs = [
+                (edge_descriptor.from_vertex, edge_descriptor.from_column),
+                (edge_descriptor.to_vertex, edge_descriptor.to_column),
+            ]
+        elif isinstance(edge_descriptor, CompositeEdgeDescriptor):
+            vertex_column_pairs = [
+                (edge_descriptor.from_vertex, from_column)
+                for from_column, _ in edge_descriptor.column_pairs
+            ] + [
+                (edge_descriptor.to_vertex, to_column)
+                for _, to_column in edge_descriptor.column_pairs
+            ]
+        else:
+            raise AssertionError(
+                f"Unknown edge descriptor type {edge_descriptor}: "
+                f"{type(edge_descriptor)} for edge {edge_name}."
```

Comment:
Same as above:
```suggestion
Did you mean to flip the type and the printout of the dataclass here?
```suggestion
                f"Unknown edge descriptor type {type(edge_descriptor)}: "
                f"{edge_descriptor} for edge {edge_name}."
```
```